File: python/linked_list.py

class Node:
    def __init__(self,value):
        self.value = value
        self.next = None


class LinkedList:
    def __init__(self):
        self.head = None

    def append(self,value):
        new_node = Node(value)
        if not self.head:
            self.head = new_node
            return
        current_node = self.head
        while current_node.next:
            current_node = current_node.next
        current_node.next = new_node

    def print_list(self):
        if not self.head:
            print("LISTA VAZIA")
            return
        current_node = self.head
        while current_node:
            print(current_node.value)
            current_node = current_node.next

File: cpp/linked_list.cpp
#include <bits/stdc++.h>

using namespace std;

class Node {
public:
    int data;
    Node* next;
};

void append(Node** head_ref, int value){

    Node* aux = new Node();
    Node *last = *head_ref;

    aux->data = value;
    aux->next = NULL;


    if(*head_ref == NULL){
        *head_ref = aux;
    }

    while ( last->next != NULL ){
        last = last->next;
    }

    last->next = aux;

    return;
}

void printList(Node* n)
{
    if (n==NULL){
        cout << "Empty List" << endl;
        return;
        }

    while (n != NULL){
        cout << n->data << " ";
          
        n = n->next;
    }
    cout << endl;  
    }

void clearList(Node* &n){
    Node* aux  = NULL;
    while (n != NULL){
        aux = n->next; 
        n = aux;
        }
    }

int main(){

    Node* head = NULL;
    Node* second = NULL;
    Node* third = NULL;

    head = new Node();
    second = new Node();
    third = new Node();


    head->data = 1;
    head->next = second;

    second->data = 2;
    second->next = third;

    third->data = 3;
    third->next = NULL;


    printList(head);
    append(&head, 4);
    printList(head);

    return 0;
}